### ISSUE TITLE
PDF "Auto" Orientation Support

### ISSUE THREAD

**Issue by tresf** [qzind/tray#158]
Take a 11.0 x 8.5 PDF, open in Chrome or Adobe:

 * Both preview the PDF as a landscape when opened
 * Both print the PDF properly in landscape when printed

Differences:
 * The PDF printed with Chrome, when printed to PDFCreator, creates a 8.5 x 11.0 rotated PDF
 * The PDF printed with Acrobat, when printed to PDFCreator, creates a 11.0 x 8.5 PDF

QZ Tray:
 * QZ Tray does not rotate the PDF when printing and seems to offer no option to auto rotate.
 * The content is boxed into a 8.5 x 11.0 cropped area, cutting off content.

Do we need a new Orientation:AUTO mode?

**Comment by tung12a1** [comment]
Please, we need it

**Comment by bberenz** [comment]
Rotated PDFs should now keep their dimensions when printed with `default` orientation under 61a2f1b9757c6aa8bc2725c645f86c0a43f163bf .

**Comment by klabarge** [comment]
Successful testing under macOS. I printed an 11 x 8.5in PDF with default values.

![image](https://cloud.githubusercontent.com/assets/12505463/26086607/b46306e2-39b9-11e7-8d04-afe7dd6112c5.png)

**Comment by tresf** [comment]
Closed via 61a2f1b.

**Comment by klabarge** [comment]
We have a client wishing to send a multi-page PDF document to QZ Tray.

The first page is 11.0 x 8.5 in 
The second page is 8.5 x 14.0 in

When printed using default orientation, both pages are printed as an 8.5 x 11.0 in page.

Let me know @bberenz if you need a reproducible document
```
[DEBUG] 2017-07-10 23:52:35,416 @ qz.ws.PrintSocketClient:?
	Message: {"call":"print","promise":{},"params":{"printer":{"name":"Microsoft Print to PDF"},"options":{"colorType":"color","copies":"1","density":"","duplex":false,"fallbackDensity":null,"interpolation":"","jobName":"","margins":"0","orientation":"","paperThickness":"","printerTray":"","rasterize":true,"rotation":"0","scaleContent":true,"size":null,"units":"in","altPrinting":false,"encoding":"","endOfDoc":"","perSpool":"1"},"data":[{"type":"pdf","data":"file:///C:/Program%20Files/QZ%20Tray/demo/assets/Test.pdf"}]},"timestamp":1499745155416,"uid":"6lg02v"}
[WARN] 2017-07-10 23:52:35,420 @ qz.ws.PrintSocketClient:?
	Bad signature on request
[INFO] 2017-07-10 23:52:36,598 @ qz.common.TrayManager:?
	Allowed localhost to print to Microsoft Print to PDF
[TRACE] 2017-07-10 23:52:36,599 @ qz.utils.PrintingUtilities:?
	Waiting for processor, 0/7 already in use
[DEBUG] 2017-07-10 23:52:36,599 @ qz.utils.PrintingUtilities:?
	Using qz.printer.action.PrintPDF to print
[DEBUG] 2017-07-10 23:52:36,599 @ qz.printer.PrintServiceMatcher:?
	Searching for PrintService matching Microsoft Print to PDF
[DEBUG] 2017-07-10 23:52:36,600 @ qz.printer.PrintServiceMatcher:?
	Found 10 printers
[DEBUG] 2017-07-10 23:52:36,600 @ qz.printer.PrintServiceMatcher:?
	Found match: Microsoft Print to PDF
[WARN] 2017-07-10 23:52:36,600 @ qz.printer.PrintOptions:?
	Cannot read  as a double for density, using default
[WARN] 2017-07-10 23:52:36,601 @ qz.printer.PrintOptions:?
	Cannot read  as a valid value for interpolation, using default
[WARN] 2017-07-10 23:52:36,601 @ qz.printer.PrintOptions:?
	Cannot read  as a valid value for orientation, using default
[WARN] 2017-07-10 23:52:36,602 @ qz.printer.PrintOptions:?
	Cannot read  as a double for paperThickness, using default
[DEBUG] 2017-07-10 23:52:36,602 @ qz.utils.PrintingUtilities:?
	Found Resolution: 60000x60000 dphi
[DEBUG] 2017-07-10 23:52:36,686 @ qz.printer.action.PrintPDF:?
	Parsed 1 files for printing
[TRACE] 2017-07-10 23:52:36,723 @ qz.printer.action.PrintPixel:?
	DPI: 600.0	CNV: 72.0
[TRACE] 2017-07-10 23:52:36,724 @ qz.printer.action.PrintPixel:?
	Drawable area: 0.0,0.0:8.5,11.0
[TRACE] 2017-07-10 23:52:36,724 @ qz.printer.action.PrintPixel:?
	[60000x60000 dphi, (0.0,0.0)->(215.9,279.4)mm, color]
[INFO] 2017-07-10 23:52:36,725 @ qz.printer.action.PrintPixel:?
	Starting printing (1 copies)
[TRACE] 2017-07-10 23:52:43,984 @ qz.printer.BookBundle:?
	Requested page 0 for printing
[DEBUG] 2017-07-10 23:52:43,985 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[TRACE] 2017-07-10 23:52:44,213 @ qz.printer.BookBundle:?
	Requested page 0 for printing
[DEBUG] 2017-07-10 23:52:44,213 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[TRACE] 2017-07-10 23:52:44,503 @ qz.printer.BookBundle:?
	Requested page 0 for printing
[DEBUG] 2017-07-10 23:52:44,504 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[TRACE] 2017-07-10 23:52:46,022 @ qz.printer.BookBundle:?
	Requested page 1 for printing
[DEBUG] 2017-07-10 23:52:46,022 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[WARN] 2017-07-10 23:52:46,095 @ org.apache.pdfbox.pdmodel.font.PDSimpleFont:325
	No Unicode mapping for .notdef (19) in font Helvetica
[WARN] 2017-07-10 23:52:46,096 @ org.apache.pdfbox.rendering.Type1Glyph2D:60
	No glyph for 19 (.notdef) in font Helvetica
[TRACE] 2017-07-10 23:52:46,135 @ qz.printer.BookBundle:?
	Requested page 1 for printing
[DEBUG] 2017-07-10 23:52:46,136 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[WARN] 2017-07-10 23:52:46,200 @ org.apache.pdfbox.rendering.Type1Glyph2D:60
	No glyph for 19 (.notdef) in font Helvetica
[TRACE] 2017-07-10 23:52:46,302 @ qz.printer.BookBundle:?
	Requested page 1 for printing
[DEBUG] 2017-07-10 23:52:46,302 @ qz.printer.PDFWrapper:?
	Paper area: 0,0:612,792
[WARN] 2017-07-10 23:52:46,360 @ org.apache.pdfbox.rendering.Type1Glyph2D:60
	No glyph for 19 (.notdef) in font Helvetica
[INFO] 2017-07-10 23:52:48,432 @ qz.utils.PrintingUtilities:?
	Printing complete
[TRACE] 2017-07-10 23:52:48,433 @ qz.utils.PrintingUtilities:?
	Returning processor back to pool
```

**Comment by bberenz** [comment]
Changes in d24cb72065ed089608174f66e9215b06b566a86f make each page a separate entity, and formatting settings are applied on a per page basis. This should allow for a variety of mixed PDF's to print correctly.

**Comment by klabarge** [comment]
Tested on Windows 10 and Mac and this works 👍 

> The first page is 11.0 x 8.5 in

This correctly prints landscape

> The second page is 8.5 x 14.0 in

This correctly prints portrait

I will need to test this to a physical printer with one tray having US Letter, one tray having Legal (8.5 x 14.0in) as the virtual printers I have do not support multiple page sizes.

**Comment by tresf** [comment]
PDF auto-orientation is now available in 2.0.5.  You can grab it at https://qz.io/download
